pangeo_forge/cli/forge.py

```diff
@@ -0,0 +1,95 @@
+import pathlib
+import runpy
+import subprocess
+
+import prefect
+import typer
+from rich.table import Table
+
+from .console import console
+
+app = typer.Typer(help="Running and validating recipes locally.")
+
+
+@app.command(help="Check that the pipeline definition is valid.")
+def check(
+    recipe: pathlib.Path = typer.Argument(
+        "recipe",
+        show_default=True,
+        exists=True,
+        dir_okay=True,
+        help="path to the recipe directory (e.g. 'recipe')",
+    )
+):
+    """
+    Check that the pipeline definition is valid. This does not run the pipeline.
+    """
+
+    errors = []
+    pipeline = recipe / "pipeline.py"
+    if not pipeline.exists():
+        errors.append(f"File '{recipe}/pipeline.py' does not exist")
+
+    else:
+        # result returns the namespace of the module as a dict of {name: value}.
+        pipeline = pipeline.as_posix()
+        result = runpy.run_path(pipeline)
+
+        if "Pipeline" not in result:
+            errors.append(f"File '{recipe}/pipeline.py' must have a class named 'Pipeline'")
+        if "flow" not in result:
+            errors.append(f"File '{recipe}/pipeline.py' must have a prefect Flow named 'flow'")
+        elif not isinstance(result["flow"], prefect.Flow):
+            errors.append(f"File '{recipe}/pipeline.py' must have a prefect Flow named 'flow'")
+        else:
+            flow = result["flow"]
+            flow.validate()
```

Comment:
It's worth checking exactly what the new recipe structure requires. Previously we needed things to be valid prefect Flows. Now I'm guessing we'd want to check the types of the various stages.